app-creator/src/widgets/widgets-tab/widgets-tab.ts

```diff
@@ -31,29 +80,106 @@ export class WidgetsTab extends LitElement {
           ></ui-label>
-        </draggable-widget>
-        <h5 class="subtitle">Button</h5>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-button',
+      name: 'button',
+      markup: html`<h5 class="subtitle">Button</h5>
         <draggable-widget .hasOverlay=${false}>
           <ui-button label="Button"></ui-button>
-        </draggable-widget>
-        <h5 class="subtitle">Select</h5>
-        <draggable-widget .hasOverlay=${false}>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-select',
+      name: 'select',
+      markup: html`<h5 class="subtitle">Select</h5>
+        <draggable-widget>
           <ui-select
             .items=${['Item 1', 'Item 2']}
             placeholder="Select Item"
           ></ui-select>
-        </draggable-widget>
-        <h5 class="subtitle">Textbox</h5>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-textbox',
+      name: 'textbox',
+      markup: html`<h5 class="subtitle">Textbox</h5>
         <draggable-widget>
           <ui-textbox label="Enter text"></ui-textbox>
-        </draggable-widget>
-        <h5 class="subtitle">Checkbox</h5>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-panel',
+      name: 'panel',
+      markup: html`<h5 class="subtitle">Panel</h5>
+        <draggable-widget>
+          <ui-panel isRaised></ui-panel>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-slider',
+      name: 'slider',
+      markup: html`<h5 class="subtitle">Slider</h5>
+        <draggable-widget>
+          <ui-slider label="Item"></ui-slider>
+        </draggable-widget>`,
+    },
+    {
+      id: 'ui-checkbox',
+      name: 'checkbox',
+      markup: html`<h5 class="subtitle">Checkbox</h5>
         <draggable-widget .hasOverlay=${false}>
           <ui-checkbox label="Item"></ui-checkbox>
-        </draggable-widget>
+        </draggable-widget>`,
+    },
+  ];
+
+  /**
+   * Sets the search query.
+   */
+  @property({ type: String }) query = '';
+
+  render() {
+    const { query, filterWidgets, handleSearch } = this;
+    const filteredWidgets: WidgetItem[] = filterWidgets(query);
+    const emptyNotice =
+      filteredWidgets.length === 0
+        ? html`
+            <div id="empty-notice">
+              <iron-icon id="empty-notice-icon" icon="search"></iron-icon>
+              <h4 id="empty-notice-message">
+                No widgets available. Please search again.
+              </h4>
+            </div>
+          `
+        : nothing;
+
+    return html`
+      <tab-container title="Widgets">
+        <search-bar @onsearch=${handleSearch}></search-bar>
+        ${filteredWidgets.map(({ markup }) => markup)} ${emptyNotice}
       </tab-container>
     `;
   }
+
+  /**
+   * Returns widgets with names and ids that include the search query.
+   */
+  filterWidgets(query: string): WidgetItem[] {
```

Comment:
Could you add a unit test to make sure matching widgets are returned given a query?